### DISCUSSION TITLE
Many-to-Many relation back to itself

### DISCUSSION THREAD

**Discussion by brandonlee781** [prisma/prisma#12605] [Q&A] [answered]
I have a database where I need to connect a model back to the same model through a link table

```prisma
model MasterVendor {
  id               Int       @id @unique(map: "id") @default(autoincrement())
  parents      vendorparents[]
}

model vendorparents {
  id                   Int       @id @unique(map: "id") @default(autoincrement())
  masterVendorId       Int?
  parentMasterVendorId Int?
  parents              MasterVendor? @relation("parents", fields: [parentMasterVendorId], references: [id])
}
```

What is supposed to happen is that the MasterVendor table will search for its own id in the vendorparent.masterVendorId field and the parent record would then be that rows parentMasterVendorId linked back to the vendor table to get any parent records. But I can't seem to figure out what I need to do with the model to get that to work.

**Comment by austincrim** [comment, marked as answer]
Hey @brandonlee781 👋 ,

I had to make a few adjustments to your schema to get it to generate, posted those below. Would you mind sharing some specifics of how you are going to use this setup in practice? Preferably with some code snippets or errors that you have run into so far.

```prisma
model MasterVendor {
  id      Int            @id @default(autoincrement())
  parents VendorParent[]
}

model VendorParent {
  id                   Int           @id @default(autoincrement())
  masterVendorId       Int?
  parentMasterVendorId Int?
  parents              MasterVendor? @relation(fields: [parentMasterVendorId], references: [id])
}
```

**Reply by brandonlee781** [reply]
I'm starting to realize that I didn't fully understand relations before I came here looking for help. I think I have it figured out and I'm going to mark as answered. Thank you.

**Comment by brandonlee781** [comment]
The intent is to have a MasterVendor be the parent or child of another MasterVendor. And MasterVendor records can have multiple parents and children. The database is already set up and in use, so I can't change anything on that end.

A sort of simplified idea is that if a MasterVendor.id exists in the VendorParent.masterVendorId column, it is a child of VendorParent.parentMasterVendorId. And if a MasterVendor.id exists in the VendorParent.parentMasterVendorId row, it is the parent of VendorParent.masterVendorId.

Using Prisma with Nexus, I'm able to connect the MasterVendor to the VendorParent type, but it MasterVendor.parents should be a list of MasterVendor, not a list of VendorParents.

**Comment by brandonlee781** [comment]
After some tinkering I was able to get the correct data with:

```ts
t.list.field('parents', {
  type: MasterVendorType,
  async resolve(record) {
    const parentIds = record?.parents?.map(p => p?.parentMasterVendorId)
    return prisma.masterVendor.findMany({
      where: {
        id: {
          in: parentIds,
        },
      },
    })
  },
})
```
But this isn't very efficient and typescript doesn't like this, because it doesn't know the parents key exists.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
